src/components/expanded-state/custom-gas/FeesPanel.js

```diff
@@ -0,0 +1,509 @@
+        10
+      );
+    }
+
+    let maxPriorityFee;
+    if (feesGweiInputFocused) {
+      // block more than 2 decimals on gwei value
+      const decimals = Number(customMaxPriorityFee) % 1;
+      maxPriorityFee =
+        `${decimals}`.length > 4
+          ? Number(parseFloat(customMaxPriorityFee).toFixed(2))
+          : customMaxPriorityFee;
+    } else {
+      maxPriorityFee = get(
+        selectedGasFee,
+        'gasFeeParams.maxPriorityFeePerGas.gwei',
+        0
+      );
+    }
+    return { currentBaseFee, maxBaseFee, maxFee, maxPriorityFee };
+  }, [
+    selectedGasFee,
+    currentBlockParams,
+    selectedOptionIsCustom,
+    feesGweiInputFocused,
+    customMaxBaseFee,
+    customMaxPriorityFee,
+  ]);
+
+  const renderRowLabel = useCallback(
+    (label, type, error, warning) => {
+      const openGasHelper = () => {
+        Keyboard.dismiss();
+        navigate(Routes.EXPLAIN_SHEET, {
+          currentBaseFee,
+          currentGasTrend,
+          type,
+        });
+      };
+      let color;
+      let text;
+      if ((!error && !warning) || !selectedOptionIsCustom) {
+        color = isDarkMode
+          ? colors.alpha(darkModeThemeColors.blueGreyDark, 0.25)
+          : colors.alpha(colors.blueGreyDark, 0.25);
+        text = '􀅵';
+      } else if (error) {
+        color = colors.red;
+        text = '􀁟';
+      } else {
+        color = colors.yellowFavorite;
+        text = '􀇿';
+      }
+
+      return (
+        <PanelColumn>
+          <ButtonPressAnimation onPress={openGasHelper}>
+            <Row>
+              <PanelLabel>
+                {`${label} `}
+                <Label color={color} size="smedium" weight="bold">
+                  {text}
+                </Label>
+              </PanelLabel>
+            </Row>
+          </ButtonPressAnimation>
+        </PanelColumn>
+      );
+    },
+    [
+      currentBaseFee,
+      currentGasTrend,
+      navigate,
+      selectedOptionIsCustom,
+      isDarkMode,
+    ]
+  );
+
+  const formattedBaseFee = useMemo(
+    () => `${parseInt(currentBaseFee, 10)} Gwei`,
+    [currentBaseFee]
+  );
+
+  const handleOnInputFocus = useCallback(() => {
+    if (isEmpty(gasFeeParamsBySpeed[gasUtils.CUSTOM])) {
+      const gasFeeParams = gasFeeParamsBySpeed[selectedGasFeeOption];
+      updateToCustomGasFee({
+        ...gasFeeParams,
+        option: gasUtils.CUSTOM,
+      });
+    } else {
+      updateGasFeeOption(gasUtils.CUSTOM);
+    }
+  }, [
+    gasFeeParamsBySpeed,
+    selectedGasFeeOption,
+    updateGasFeeOption,
+    updateToCustomGasFee,
+  ]);
+
+  const handleFeesGweiInputFocus = useCallback(() => {
+    onCustomGasFocus?.();
+    handleOnInputFocus();
+    const {
+      gasFeeParams: { maxFeePerGas, maxPriorityFeePerGas },
+    } = selectedGasFee;
+    setCustomMaxPriorityFee(get(maxPriorityFeePerGas, 'gwei', 0));
+    setCustomMaxBaseFee(parseInt(get(maxFeePerGas, 'gwei', 0), 10));
```

Comment:
is this `parseInt` needed here?

Reply:
because we're displaying int for base fee

Reply:
conclusion: use toFixedDecimals so that it does rounding instead of just truncating decimals